Add eslint for payment promotion row

louwie17 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Add ESLint for `dangerouslySetInnerHTML` case.

Context: pcShBQ-2Ih-p2

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

N/A

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

louwie17 committed:
Add eslint for dangerouslySetInnerHTML for subTitleContent

louwie17 committed:
Add changelog

roykho commented on plugins/woocommerce/client/admin/client/wp-admin-scripts/payment-method-promotions/payment-promotion-row.tsx:
```diff
@@ -151,6 +151,7 @@ export const PaymentPromotionRow: React.FC< PaymentPromotionRowProps > = ( {
 								subTitleContent ? (
 									<div
 										className="pre-install-payment-gateway__subtitle"
+										// eslint-disable-next-line react/no-danger - innerHTML from the element with class name: gateway-subtitle.
```
I think you're missing a hyphen here? Should be `--`?

louwie17 replied:
Good catch, updated here: https://github.com/woocommerce/woocommerce/pull/54119/commits/84f51fcb67a4bef06e046b82209342e4ebf3fe53

louwie17 committed:
Add additional hyphen

roykho reviewed (changes requested):
I am still seeing failing CI, try updating the branch. The other thing I just want to point out is that perhaps we don't need to add a changelog here because this is not public facing change. So just a comment style changelog entry will do.

louwie17 committed:
Move changelog to a comment

louwie17 commented:
> I am still seeing failing CI, try updating the branch. The other thing I just want to point out is that perhaps we don't need to add a changelog here because this is not public facing change. So just a comment style changelog entry will do.

Thanks @roykho, I didn't realize my branch was so out of date. I just rebased and also updated the changelog to a comment, thanks!

roykho reviewed (approved):
LGTM!
